api/v1beta1/v1beta2_condition_consts.go

```diff
@@ -0,0 +1,262 @@
+	ClusterControlPlaneAvailableV1Beta2Condition = "ControlPlaneAvailable"
+
+	// ClusterWorkersAvailableV1Beta2Condition is the summary of MachineDeployment and MachinePool's Available conditions.
+	ClusterWorkersAvailableV1Beta2Condition = "WorkersAvailable"
+
+	// ClusterMachinesReadyV1Beta2Condition surfaces detail of issues on the controlled machines, if any.
+	ClusterMachinesReadyV1Beta2Condition = MachinesReadyV1beta2Condition
+
+	// ClusterMachinesUpToDateV1Beta2Condition surfaces details of Cluster's machines not up to date, if any.
+	ClusterMachinesUpToDateV1Beta2Condition = MachinesUpToDateV1beta2Condition
+
+	// ClusterRemoteConnectionProbeV1Beta2Condition is true when control plane can be reached; in case of connection problems.
+	// The condition turns to false only if the cluster cannot be reached for 50s after the first connection problem
+	// is detected (or whatever period is defined in the --remote-connection-grace-period flag).
+	ClusterRemoteConnectionProbeV1Beta2Condition = "RemoteConnectionProbe"
+
+	// ClusterScalingUpV1Beta2Condition is true if available replicas < desired replicas.
+	ClusterScalingUpV1Beta2Condition = ScalingUpV1beta2Condition
+
+	// ClusterScalingDownV1Beta2Condition is true if replicas > desired replicas.
+	ClusterScalingDownV1Beta2Condition = ScalingDownV1beta2Condition
+
+	// ClusterRemediatingV1Beta2Condition surfaces details about ongoing remediation of the controlled machines, if any.
+	ClusterRemediatingV1Beta2Condition = RemediatingV1beta2Condition
+
+	// ClusterDeletingV1Beta2Condition surfaces details about ongoing deletion of the cluster.
+	ClusterDeletingV1Beta2Condition = DeletingV1beta2Condition
+
+	// ClusterPausedV1Beta2Condition is true if Cluster and all the resources being part of it are paused.
+	ClusterPausedV1Beta2Condition = PausedV1beta2Condition
+)
+
+// Conditions that will be used for the MachineHealthCheck object in v1Beta2 API version.
+const (
+	// MachineHealthCheckReadyV1beta2Condition is true if the MachineHealthCheck is not deleted,
+	// and RemediationAllowed condition is true.
+	MachineHealthCheckReadyV1beta2Condition = ReadyV1beta2Condition
+
+	// MachineHealthCheckRemediationAllowedV1beta2Condition surfaces whether the MachineHealthCheck is
+	// allowed to remediate any Machines or whether it is blocked from remediating any further.
+	MachineHealthCheckRemediationAllowedV1beta2Condition = "RemediationAllowed"
+
+	// MachineHealthCheckPausedV1Beta2Condition is true if this MachineHealthCheck or the Cluster it belongs to are paused.
+	MachineHealthCheckPausedV1Beta2Condition = PausedV1beta2Condition
+)
+
+// Conditions that will be used for the ClusterClass object in v1Beta2 API version.
+const (
+	// ClusterClassReadyV1beta2Condition is true if the ClusterClass is not deleted,
+	// and both VariablesReady and ReferencedVersionsUpToDate conditions are true.
+	ClusterClassReadyV1beta2Condition = ReadyV1beta2Condition
```

Comment:
Looks like we don't have Ready on the CC today, do we really want to introduce it now?
I would prefer to have this as a follow-up if we think we really need a Ready condition on the CC

Reply:
Dropped